tests/function-controller/testsuite/testsuite.go

```diff
@@ -106,62 +133,94 @@ func (t *TestSuite) Run() {
 
 	t.t.Log("Creating function...")
 	functionDetails := t.getFunction(helloWorld)
-	fnResourceVersion, err := t.function.Create(functionDetails)
+	err = t.function.Create(functionDetails)
 	failOnError(t.g, err)
 
 	t.t.Log("Waiting for function to have ready phase...")
-	err = t.function.WaitForStatusRunning(fnResourceVersion)
+	err = t.function.WaitForStatusRunning()
+	failOnError(t.g, err)
+
+	t.t.Log("Creating addons configuration...")
+	err = t.addonsConfig.Create(addonsConfigUrl)
+	failOnError(t.g, err)
+
+	t.t.Log("Waiting for addons configutation to have ready phase...")
+	err = t.addonsConfig.WaitForStatusRunning()
+	failOnError(t.g, err)
+
+	t.t.Log("Creating service instance...")
+	err = t.serviceinstance.Create(serviceClassExternalName, servicePlanExternalName)
+	failOnError(t.g, err)
+
+	t.t.Log("Waiting for service instance to have ready phase...")
+	err = t.serviceinstance.WaitForStatusRunning()
+	failOnError(t.g, err)
+
+	t.t.Log("Creating service binding...")
+	err = t.servicebinding.Create(t.cfg.ServiceInstanceName)
+	failOnError(t.g, err)
+
+	t.t.Log("Waiting for service binding to have ready phase...")
+	err = t.servicebinding.WaitForStatusRunning()
+	failOnError(t.g, err)
+
+	t.t.Log("Creating service binding usage...")
+	// we are deliberately creating servicebindingusage HERE, to test how it behaves after function update
+	err = t.servicebindingusage.Create(t.cfg.ServiceBindingName, t.cfg.FunctionName, t.cfg.UsageKindName, redisEnvPrefix)
```

Comment:
You can put UsageKind as one of field in SBU struct, and not pass UsageKInd in Create function.